mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -434,6 +434,14 @@ def init_conductors(self):
         mwxrun.grid.potential_zmin = self.cathode.V
         mwxrun.grid.potential_zmax = self.anode.V
 
+        # Set particles to be scraped
+        if hasattr(self, 'electrons'):
+            self.electrons.scrape_zmin = 1
+            self.electrons.scrape_zmax = 1
+        if hasattr(self, 'ions'):
+            self.ions.scrape_zmin = 1
+            self.ions.scrape_zmax = 1
+
     def init_scraper(self):
```

Comment:
Do we still need this function then?

Reply:
Looks like there's an input parameter for picmi species to set this called `warpx_save_particles_at_blank` where blank is one of `xlo`, `xhi`, `zlo`, `zhi`, `ylo`, `yhi`, or `eb`. So these values can be passed in when the species is initialized.

But how about we add another input parameter for diode setup which is a list of strings for which boundaries to scrape? Then we use those to set all those `warpx_save_particles_at_blank` parameters.

Reply:
We could reorganize things such that the `init_scraper` function will set the `scrape_zmin` and `scrape_zmax` values for each species. This is probably a better approach since if we activate the scraping but don't install a flux diagnostic the particle buffer will continue to grow indefinitely. So the best is probably to have the `init_fluxdiag` function call the `init_scraper` function.

Reply:
So should I add the input parameter I mentioned so that we can choose what boundaries to scrape from or just keep it hardcoded to xlo/hi like it is now?

Reply:
@MKieburtz Since in a diode we will always have exactly a cathode and anode there is no need to add the option of scraping in the x-direction. But please do make the change to move lines 437-443 to be inside the `init_scraper` function and add a call to that function to the `init_fluxdiag` function.